hub/gads-ui/src/components/DeviceControl/StreamCanvas/StreamCanvas.js

```diff
@@ -39,43 +48,242 @@ export default function StreamCanvas({ deviceData, shouldShowStream }) {
 
     const handleOrientationButtonClick = (isPortrait) => {
         setIsPortrait(isPortrait)
+        updateCanvasDimensions(isPortrait)
     }
 
+    // Handles orientation/resizing only
+    useEffect(() => {
+        const handleResize = () => {
+            // Only update if the ref still exists
+            if (videoDimensionsRef.current) {
+                updateCanvasDimensions(isPortrait)
+            }
+        }
+
+        updateCanvasDimensions(isPortrait)
+
+        window.addEventListener('resize', handleResize)
+        return () => window.removeEventListener('resize', handleResize)
+    }, [])
```

Comment:
Add isPortrait to the dependency array to ensure it correctly updates when the orientation changes.

Reply:
This useEffect is only to register the event listener and to update the dimensions on initial load - we separate update the dimensions in the `handleOrientationButtonClick` function